tests/run-fedora

```diff
@@ -23,12 +23,17 @@ mkdir -p /run/systemd/system
 
 # run build and test as user
 useradd build
-su -s /bin/sh - build << EOF
+su -s /bin/sh - build << EOF || { [ -z "$DEBUG" ] || sleep infinity; exit 1; }
 set -ex
 cd /source
 export TEST_CODE="$TEST_CODE"
-python3 -m unittest -v || {
-  [ -z "$DEBUG" ] || sleep infinity
-  exit 1
-}
+
+python3 -m unittest -v
+
+# run all tests with pytest only on TEST_CODE, as that is expensive
+if [ -n "$TEST_CODE" ]; then
+    python3 -m pytest -v
+else
+    python3 -m pytest -vv -k 'test_pytest or TestAPI'
```

Comment:
fwiw, if you install `python3-pytest-xdist` you can run with `pytest -n auto ...` and it'll run the tests in parallel to speed things up.